record.go

```diff
@@ -0,0 +1,444 @@
+package libdns
+
+import (
+	"fmt"
+	"net/netip"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// Record is any type that can reduce itself to the [RR] struct.
+type Record interface {
+	RR() (RR, error)
+}
+
+// RR represents a [DNS Resource Record], which resembles how records are
+// represented by DNS servers in zone files.
+//
+// The fields in this struct are common to all RRs, with the data field
+// being opaque; it has no particular meaning until it is parsed.
+//
+// [DNS Resource Record]: https://en.wikipedia.org/wiki/Domain_Name_System#Resource_records
+type RR struct {
+	// The name of the record. It is partially qualified, relative to the zone.
+	// For the sake of consistency, use "@" to represent the root of the zone.
+	// An empty name typically refers to the last-specified name in the zone
+	// file, which is only determinable in specific contexts.
+	//
+	// (For the following examples, assume the zone is “example.com.”)
+	//
+	// Examples:
+	//   - “www” (for “www.example.com.”)
+	//   - “@” (for “example.com.”)
+	//   - “subdomain” (for “subdomain.example.com.”)
+	//   - “sub.subdomain” (for “sub.subdomain.example.com.”)
+	//
+	// Invalid:
+	//   - “www.example.com.” (fully-qualified)
+	//   - “example.net.” (fully-qualified)
+	//   - "" (empty)
+	//
+	// Valid, but probably doesn't do what you want:
+	//   - “www.example.net” (refers to “www.example.net.example.com.”)
+	Name string
+
+	// The time-to-live of the record. This is represented in the DNS zone file as
+	// an unsigned integral number of seconds, but is provided here as a
+	// [time.Duration] for ease of use in Go code. Fractions of seconds will be
+	// rounded down (truncated). A value of 0 means that the record should not be
+	// cached. Some provider implementations may assume a default TTL from 0; to
+	// avoid this, set TTL to a sub-second duration.
+	//
+	// Note that some providers may reject or silently increase TTLs that are below
+	// a certain threshold, and that DNS resolvers may choose to ignore your TTL
+	// settings, so it is recommended to not rely on the exact TTL value.
+	TTL time.Duration
+
+	// The type of the record as an uppercase string. DNS provider packages are
+	// encouraged to support as many of the most common record types as possible,
+	// especially: A, AAAA, CNAME, TXT, HTTPS, and SRV.
+	//
+	// Other custom record types may be supported with implementation-defined
+	// behavior.
+	Type string
+
+	// The data (or "value") of the record. This field should be formatted in the
+	// *unescaped* standard zone file syntax (technically, the "RDATA" field as
+	// defined by RFC 1035 §5.1). Due to variances in escape sequences and provider
+	// support, this field should not contain escapes.
```

Comment:
Just wondering: What kind of escapes are meant here?
For example https://datatracker.ietf.org/doc/html/rfc3597 has RDATA values with what I would think are escape sequences, e.g. `e.example.   IN          A               \# 4 0A000001`

Reply:
Hmm, well I think since escape sequences can be "parsed" or "evaluated" to become their _actual_ values, we intend that this field always contains the parsed/evaluated contents. So, maybe _any_ escapes? I dunno though... @gucci-on-fleek might have good thoughts on this.

Reply:
I think it sounds good and the godoc looks good concerning this now, I'll resolve this but feel free to continue discussion :+1: